### PR TITLE
en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2: ad…

### PR DESCRIPTION
**Pull request by amathuria**
…d blog post about mClock vs WPQ testing with emphasis on scrub operations

Part 2 of mClock vs WPQ testing with background operations

Signed-off-by: Aishwarya Mathuria <amathuri@redhat.com>

### CODE REVIEW

File: src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md
Comment (ljflores) [inline src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md:123]:
```suggestion
To get a better understanding of what the client latencies look like while scrubbing is going on, we can refer to the graph below where we can see that high_client_ops has lower latency than WPQ throughout the duration of the test
```

File: src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md
Comment (ljflores) [inline src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md:130]:
```suggestion
From the initial testing, it was evident that WPQ finishes a deep scrub faster than any of the default mClock profiles. After taking a deeper look into what could be contributing to this, we realized that mClock treats a scheduled scrub and a requested scrub in a similar manner, whereas WPQ does not. The priority given to a [scheduled scrub](https://docs.ceph.com/en/quincy/rados/configuration/osd-config-ref/#confval-osd_scrub_priority) is 5, whereas the priority for a [requested scrub](https://docs.ceph.com/en/quincy/rados/configuration/osd-config-ref/#confval-osd_requested_scrub_priority) is 120! As the test triggers a requested scrub, the difference between WPQ scrub time and mClock scrub time was considerable. In order to continue using the test to compare scheduled deep scrubs , we set the osd_requested_scrub_priority to 5, which is the priority of a scheduled scrub. All tests below use this value.
```

File: src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md
Comment (ljflores) [inline src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md:169]:
```suggestion
The idea behind this experiment is to find out which existing scrub parameters can be tuned to finish scrub faster and how they affect client latency and throughput.
```

File: src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md
Comment (ljflores) [inline src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md:196]:
```suggestion
In conclusion, this experiment proves that increasing osd_max_scrubs to reduce scrub duration is a viable option, but that the user should be careful not to increase the concurrent scrubs to a very high number since this affects the client throughput and latency.
```

File: src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md
Comment (ljflores) [inline src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md:200]:
```suggestion
Every operation has a cost associated with it. As discussed in the beginning of this section, the lower the cost of the operation, the higher are the chances of it getting dequeued faster. The scrub operations, by default, have a cost of 50MB.
```

File: src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md
Comment (ljflores) [inline src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md:201]:
```suggestion
WPQ uses a combination of priority and cost to decide which operations are dequeued. Due to this, it is easy to control operations such as scrub. For example, the osd_requested_scrub_priority is 120. This means that when a user explicitly asks for a scrub instead of waiting for the scheduled scrub, WPQ is able to prioritize scrub operations over client operations (priority of 63). We are looking at the cost parameter with the hope that it could do the same for mClock.
```

File: src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md
Comment (ljflores) [inline src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md:203]:
```suggestion
For all the mClock tests mentioned below, we have reduced the scrub cost to 20MB. For the WPQ tests, we have used the default osd_requested_scrub_priority of 120.
```

File: src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md
Comment (ljflores) [inline src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md:226]: This made me think of the Simon and Garfunkel song "Bridge over Troubled Water" -- the lyrics go:
"When you're down and out
When you're on the street..."

;)

File: src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md
Comment (ljflores) [inline src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md:241]:
```suggestion
The graph above displays the drop in client throughput with recovery and scrub happening in parallel. All the profiles have a similar drop in throughput, and the difference between them is not significant. To get a better understanding, we will compare the client latencies in the next sections.
```

File: src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md
Comment (ljflores) [inline src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md:250]:
```suggestion
Over time, this spike flattens out, and we can see client latency returning to a normal trend-- at times even lower than WPQ.
```

File: src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md
Comment (ljflores) [inline src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md:265]:
```suggestion
In addition to this, the experiments with the mClock custom profile proved that with higher allocations, we can achieve a faster scrub. Other than mClock parameters, we also explored the existing Ceph configuration parameters that can be tuned to give us better scrub performance with mClock.
```

File: src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md
Comment (ljflores) [inline src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md:266]:
```suggestion
The mClock scheduler is the default osd_op_queue in the Quincy release. Let us know how your experience with mClock goes, and thank you for reading!
```

You can also include a sentence about reaching out to <ceph-users@ceph.io> if anyone has questions.

File: src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md
Comment (ljflores) [inline src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md:264]: Is this a way of saying that mClock profiles attempt to prioritize recovery operations over scrub? I am just a little unclear about what this sentence means.

**Review by ljflores — COMMENTED** [review]
Looks great! I added some comma/grammar suggestions, and a few other minor suggestions about clarification or details to add.

**Comment by ljflores** [comment]
Also there is still an issue with the site not compiling correctly. I looked into the details, but it's tough to say where the error lies in relation to your changes.

**Comment by amathuria** [comment]
> Also there is still an issue with the site not compiling correctly. I looked into the details, but it's tough to say where the error lies in relation to your changes.

@ljflores thanks a lot for the review! I have been trying to fix the compiling issue as well thanks for taking a look.

File: src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md
Comment (ljflores) [inline src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md:8]: @amathuria Sridhar had the same compilation issue on his blog post. We found out that it's because all tags need to have lowercase letters!

```suggestion
  - "mclock"
```

File: src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md
Comment (amathuria) [reply src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md:264]: Thanks! I'll work on rephrasing it

Comment (amathuria) [reply src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md:269]: Instead of adding more details to the conclusion, I have added some explanation in the section before. Let me know if that looks better now.

Comment (ljflores) [reply src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md:269]: Looks great!

File: src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md
Comment (ljflores) [inline src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md:271]: Period after ceph-users@ceph.io
```suggestion
The mClock scheduler is the default osd_op_queue in the Quincy release. Let us know how your experience with mClock goes, and thank you for reading! For any queries please do reach out to ceph-users@ceph.io.
```

File: src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md
Comment (ljflores) [inline src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md:73]: The formatting looks a little off here (Device Type is right next to Number of OSD). I think Sridhar fixed this in his post by adding a line break.

File: src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md
Comment (ljflores) [inline src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md:123]: Period after "duration of the test"
```suggestion
To get a better understanding of what the client latencies look like while scrubbing is going on, we can refer to the graph below where we can see that high_client_ops has lower latency than WPQ throughout the duration of the test.
```

File: src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md
Comment (ljflores) [inline src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md:173]: Maybe put a link for this config option here? https://github.com/ceph/ceph/blob/97d9de3340864c6ae55ba59cd7c8ea31f2a5cda0/src/common/options/global.yaml.in#L3607-L3614

**Review by ljflores — APPROVED** [review]
Looks good! Had a couple more follow-up reviews. Also, if you want, you can add a title image to help the blog post stand out more. This is not a "must have" though.

Blog post rendered here: https://wip-amathuria-mclock-wpq-study-blog-part2.ceph.io/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/

File: src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md
Comment (ljflores) [inline src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md:2]: Sridhar's title is "QoS in Quincy: Comparison of mClock and WPQ schedulers with background operations - Part 1", so we can modify this one to match it:
```suggestion
title: "QoS in Quincy: Comparison of mClock and WPQ schedulers with background and recovery operations - Part 2"
```

File: src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md
Comment (ljflores) [inline src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md:13]: Link to Sridhar's blog post when it is merged

**Review by sseshasa — APPROVED** [review]
LGTM!

**Comment by ljflores** [comment]
Looks great! I'd say we are ready to merge. I will leave it to @Thingee or @neha-ojha.

**Comment by ljflores** [comment]
@amathuria let's update the date to today, and I will merge it if no one has objections

**Comment by amathuria** [comment]
> @amathuria let's update the date to today, and I will merge it if no one has objections

@ljflores I'll do that thanks!

**Pull request merged**
